lib/model/query/forms.js

```diff
@@ -172,6 +172,14 @@ const del = (form) => ({ run, Assignments }) =>
   Promise.all([ run(markDeleted(form)), Assignments.revokeByActeeId(form.acteeId) ]);
 del.audit = (form) => (log) => log('form.delete', form);
 
+const restore = (form) => ({ run, Forms }) =>
+  Forms.getByProjectAndXmlFormId(form.projectId, form.xmlFormId)
+    .then((activeForm) => (activeForm.isDefined()
+      ? reject(Problem.user.uniquenessViolation({ fields: 'xmlFormId', values: form.xmlFormId }))
+      : run(markUndeleted(form))));
+restore.audit = (form) => (log) => log('form.restore', form);
```

Comment:
I think it makes sense for there to be a new audit action. For it to show up in Frontend, it'll also need to be added to the nonverbose list in `lib/model/query/audits.js`. (Frontend will also need to be modified to display the action nicely in the audit log and to show it in the filter.)

Reply:
yes, will add this verb for admins and project managers